olive/cache.py

```diff
@@ -165,6 +166,44 @@ def get_local_path(resource_path: Optional[ResourcePath], cache_dir: Union[str,
         return download_resource(resource_path, cache_dir).get_path()
 
 
+def normalize_data_path(data_root: Union[str, Path], data_dir: Union[str, Path, ResourcePath]):
+    """
+    Normalize data path, if data_dir is absolute path, return data_dir, else return data_root/data_dir
+    """
+    if isinstance(data_dir, ResourcePath):
+        data_dir_str = data_dir.get_path()
+    else:
+        data_dir_str = data_dir
+
+    data_full_path = None
+    if not data_dir_str:
+        data_full_path = data_root
+    elif Path(data_dir_str).is_absolute():
+        data_full_path = data_dir_str
+    else:
+        if data_root:
+            if isinstance(data_dir, ResourcePath) and data_dir.is_azureml_resource():
+                raise ValueError("could not append AzureML data to data_root")
+
+            # we cannot use Path to join the path. If the data_root is something like: azureml://, then Path will
```

Comment:
Does this case mean data_root an AML data, and data_dir is the relative path to AML data root path?

Reply:
Based on Mike's design doc.
The logic will be like:
1. if data_root is given, but data_dir not, just use data_root.
2. if data_root is not given, but data_dir is, use data_dir.
3. When both of them are set, concatenate as data_root / data_dir.

In this way, there won't be too many absolute data path.

Reply:
I mean this logic works for both local data and AML data?

Also, does this logic works if the user wants to use both AML data and local data? Seems data_root can only be one of them?

Reply:
If user want to use both AML data and local data at the same time, he need to specify the absolute local path.

Reply:
the documentation demonstrate how to use both of them